graph.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/python3
+# graph.py
+from collections import defaultdict
+
+
+class Graph():
+    def __init__(self, vertices):
+        self.graph = defaultdict(list)
+        self.V = vertices
+
+    def add_edge(self, u, v):
+        self.graph[u].append(v)
+    #internal
+    def is_cyclic_util(self, v, visits_p,n):
+        visits=list(visits_p)
+        visits[v] +=1
+        bads=[]
+        if(visits[v]>1):
+            if(not(v==n)):
+                visits[v] +=-1
+                return []
+            return [[v]]
+            bads.append(v)
+        for neighbor in self.graph[v]:
+            cyclic_data=self.is_cyclic_util(neighbor, visits,n)
+            if len(cyclic_data)>0:
+                for bad in cyclic_data:
+                    if(v in bad):
+                        bad.remove(v)
+                    b=[v]+bad
+                    if(not(b in bads)):
+                        bads.append(b)
+
+        return bads
+    
+    def normalize_cycle(self,a):
```

Comment:
A single line indicating what this function does would be useful.

Reply:
(the binary spacing is also mildly infuriatingly off ;)